packages/lib/services/search/SearchEngine.ts

```diff
@@ -59,7 +59,7 @@ export interface ComplexTerm {
 	value: string;
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 	scriptType: any;
-	valueRegex?: RegExp;
+	valueRegex?: string;
```

Comment:
`valueRegex` is currently set to the result of `SearchEngine.queryTermToRegex`, which returns a `string` and not a `RegExp`.